src/tests/ftest/launch.py

```diff
@@ -136,8 +136,11 @@ def set_test_environment():
         with open(os.path.join(net_path, device, "operstate"), "r") as buffer:
             state = buffer.read().strip()
         # Get the interface speed - used to select the fastest available
-        with open(os.path.join(net_path, device, "speed"), "r") as buffer:
-            speed = int(buffer.read().strip())
+        if state.lower != "up":
+            speed = 0
+        else:
+            with open(os.path.join(net_path, device, "speed"), "r") as buffer:
+                speed = int(buffer.read().strip())
```

Comment:
I wonder if this would not be more appropriate:
```
                try:
                    speed = int(buffer.read().strip())
                except IOError as error:
                    print(    -- Error obtaining speed for {} (defaulting to 0): {}".format(device, error)
                    speed = 0
```

Reply:
I am not sure... the output will get a bit messy...  Do you really want to see the IOError for any network that is down?

  - eth0  (speed:  10000 state: up)
      -- Error obtaining speed for eth1 (defaulting to 0):  IOError: [Errno 22] Invalid argument
   - eth1 (speed: 0 state: down)
   - ib0 (speed: 100000 state: up)
With the change above the ouput looks like this:
   - eth0 (speed: 10000 state: up)
   - eth1 (speed: 0 state: down)
   - ib0 (speed: 100000 state: up)